frontend/containers/forms/uploader/component.tsx

```diff
@@ -0,0 +1,183 @@
+import { useEffect, useState, useCallback } from 'react';
+
+import { FileRejection, useDropzone } from 'react-dropzone';
+import { FieldValues, Path } from 'react-hook-form';
+import { FormattedMessage } from 'react-intl';
+
+import classNames from 'classnames';
+
+import Image from 'next/image';
+
+import { ProjectImageGallery } from 'types/project';
+
+import { directUpload } from 'services/direct-upload/directUpload';
+
+import { UploaderProps } from './types';
+
+export const bytesToMegabytes = (bytes: number): number => {
+  return bytes / 1000000;
+};
+
+export const FILE_UPLOADER_MAX_SIZE = 1500000;
+
+export const Uploader = <FormValues extends FieldValues>({
+  fileTypes,
+  maxFiles = 1,
+  maxSize = FILE_UPLOADER_MAX_SIZE,
+  disabled = false,
+  register,
+  onUpload,
+  setError,
+  clearErrors,
+}: UploaderProps<FormValues>) => {
+  const [isUploading, setIsUploading] = useState<boolean>(false);
+  const [filesLength, setFilesLength] = useState<number>(0);
+
+  const setMaxSizeError = useCallback(
+    (file: File) => {
+      setError('project_images_attributes', {
+        message: (
+          <FormattedMessage
+            defaultMessage="File {fileName} is larger than {fileSize}MB"
+            id="v9GdgG"
+            values={{
+              fileName: file.name,
+              fileSize: bytesToMegabytes(maxSize),
+            }}
+          />
+        ) as any,
+      });
+    },
+    [maxSize, setError]
+  );
+
+  const uploadFiles = useCallback(
+    async (files: File[]) => {
+      if (!files.length) return;
+      setFilesLength(files.length);
+      setIsUploading(true);
+
+      const uploadedFiles: ProjectImageGallery[] = [];
+
+      await Promise.all(
+        files.map(async (file) => {
+          if (file.size > maxSize) {
+            setMaxSizeError(file);
+          } else {
+            try {
+              const { signed_id, filename, direct_upload } = await directUpload(file);
+              uploadedFiles.push({
+                id: signed_id,
+                file: signed_id,
+                cover: false,
```

Comment:
Could you make the first image a cover image by default? This way the user will know they can select one. At the moment, nothing indicate they can choose the cover image.